libass/ass_drawing.c

```diff
@@ -189,46 +211,47 @@ static ASS_DrawingToken *drawing_tokenize(const char *str)
                 if (!m_seen)
                     return NULL;
                 if (!add_root_node(&root, &tail, &points, point, TOKEN_MOVE_NC))
-                    continue;
+                    return NULL;
             }
-            points += add_many_points(&p, &tail, TOKEN_MOVE_NC, 1);
+            points += add_many_points(&p, &tail, TOKEN_MOVE_NC, 1, &error);
             break;
         case 'l':
             if (!root)
                 continue;
-            points += add_many_points(&p, &tail, TOKEN_LINE, 1);
+            points += add_many_points(&p, &tail, TOKEN_LINE, 1, &error);
             break;
         case 'b':
             if (!root)
                 continue;
-            points += add_many_points(&p, &tail, TOKEN_CUBIC_BEZIER, 3);
+            points += add_many_points(&p, &tail, TOKEN_CUBIC_BEZIER, 3, &error);
             break;
         case 's':
             if (!root)
                 continue;
             // Only the initial 3 points are TOKEN_B_SPLINE,
             // all following ones are TOKEN_EXTEND_SPLINE
             spline_start = tail;
-            size_t new_points = add_3_points(&p, &tail, TOKEN_B_SPLINE);
+            size_t new_points = add_3_points(&p, &tail, TOKEN_B_SPLINE, &error);
             points += new_points;
-            if (new_points == 0) {
+            if (new_points == 0 || error) {
                 spline_start = NULL;
-                continue;
+                break;
             }
             //-fallthrough
         case 'p':
             if (points < 3)
                 continue;
             // XXX: use TOKEN_EXTEND_SPLINE
-            points += add_many_points(&p, &tail, TOKEN_B_SPLINE, 1);
+            points += add_many_points(&p, &tail, TOKEN_B_SPLINE, 1, &error);
             break;
         case 'c':
             if (!spline_start)
                 continue;
             // Close b-splines: add the first three points of the b-spline back to the end
             for (int i = 0; i < 3; i++) {
                 // XXX: use TOKEN_EXTEND_SPLINE
-                add_node(&tail, TOKEN_B_SPLINE, spline_start->point);
+                if (!add_node(&tail, TOKEN_B_SPLINE, spline_start->point))
+                    goto fail;
```

Comment:
I think it's better to reuse `error` approach here to homogenize control flow. In that case error handling can be put directly under `if (error) { ... }` after the switch.

Reply:
If you mean manually setting `error = 1` inside the `if` block, then it’s a bit more verbose but still fine so sure.  
If you meant `add_node` should take and set an `error` parameter, then i don’t think this is a good idea as we’ll require an `error` variable also in `add_*_points`

Reply:
> If you mean manually setting `error = 1` inside the `if` block

Yes, that.

Reply:
done